src/main/scala/com/fulcrumgenomics/fgsv/util/GenomicRange.scala

```diff
@@ -0,0 +1,27 @@
+package com.fulcrumgenomics.fgsv.util
+
+import htsjdk.samtools.util.CoordMath
+
+/** A genomic interval and useful methods. */
+case class GenomicRange(refIndex: Int, start: Int, end: Int) extends Ordered[GenomicRange] {
+  require(start <= end)
+
+  /** Returns if the other interval overlaps this one. */
+  def overlaps(other: GenomicRange): Boolean = {
+    this.refIndex == other.refIndex && CoordMath.overlaps(this.start, this.end, other.start, other.end)
+  }
+
+  /** Returns the union of the two genomic intervals.  The intervals must overlap. */
+  def union(other: GenomicRange): GenomicRange = {
+    require(this.overlaps(other))
```

Comment:
Maybe be kind and emit the intervals in the message?

Reply:
todo